### PR TITLE
Remove v1 profile labels

### PR DESCRIPTION
**Pull request by neiljerram**
1. Felix Calculation Graph: use v3 Profile instead of v1 ProfileLabels
2. Felix Syncer: stop sending v1 profile labels, as no longer needed
3. Remove ProfileLabelsKey infrastructure, which is now unused

### CODE REVIEW

**Comment by caseydavenport** [comment]
Strictly speaking this is going to introduce an incompatibility with old versions of Typha I believe, but those versions should be sufficiently old enough now that I don't think it matters.

Still worth a quick spin with a mismatched version (e.g., off by one minor version) if we don't already have such an automated test case.

**Comment by neiljerram** [comment]
@caseydavenport Thanks, good thought.  It should be good because we have been streaming the whole v3.Profile resource (as well as the v1 pieces) for a long time now.  Do you know what options or existing setup we have for running that kind of version skew test?

**Comment by caseydavenport** [comment]
@neiljerram whoops! Missed your comment.

I'm not sure what options there are besides installing an old version of Typha and a new version of calico/node on a cluster by hand. I think it's safe to merge this without that test though, since as you say v3 profiles have been coming across for a long time now.

I don't even know what sort of version skew you'd need to test, but it's probably not supported anyway now that I think about it :laughing:

**Review by caseydavenport — APPROVED** [review]

File: felix/labelindex/label_inheritance_index.go [hunk 1]
```diff
@@ -148,10 +150,13 @@ func (l *InheritIndex) OnUpdate(update api.Update) (_ bool) {
 			log.Debugf("Deleting host endpoint %v from InheritIndex", key)
 			l.DeleteLabels(key)
 		}
-	case model.ProfileLabelsKey:
+	case model.ResourceKey:
```
Comment (robbrockbank) [inline felix/labelindex/label_inheritance_index.go:153]: According to this change, older versions of Felix will still be expecting the ProfileLabelsKey to be sync'd from Typha.

File: libcalico-go/lib/backend/syncersv1/updateprocessors/profileprocessor.go [hunk 2]
```diff
@@ -77,25 +76,18 @@ func (pup *profileUpdateProcessor) Process(kvp *model.KVPair) ([]*model.KVPair,
 		}
 	}
```
Comment (robbrockbank) [inline libcalico-go/lib/backend/syncersv1/updateprocessors/profileprocessor.go:78]: Theses changes mean an up level Typha will stop syncing the old v1 Labels key/values.

But older versions of Felix still use the v1 Labels key/values for label inheritance.

File: felix/labelindex/label_inheritance_index.go (hunk 1 above)
Comment (neiljerram) [reply felix/labelindex/label_inheritance_index.go:153]: Addressed by #5695

File: libcalico-go/lib/backend/syncersv1/updateprocessors/profileprocessor.go (hunk 2 above)
Comment (neiljerram) [reply libcalico-go/lib/backend/syncersv1/updateprocessors/profileprocessor.go:78]: Addressed by #5695

**Pull request merged**